packages/jest-runner/src/run_test.js

```diff
@@ -98,6 +93,13 @@ async function runTestInternal(
     testConsole = new BufferedConsole();
   }
 
+  const environment = new TestEnvironment(
+    Object.assign({}, config, {console: testConsole}),
```

Comment:
Can you split this up into separate options and pass two arguments instead? Let's not add the console to the main config object of Jest.

Reply:
Didn't we add an environmentOptions recently? Can we reuse that?

Reply:
I thought you might ask that (proof: watch the video on my twitch, I live streamed myself making this pr :smile:). But decided against it because I thought that an other argument might be less desirable considering the TestEnvironment can be many things, not just the jsdom one. However, if there's an `environmentOptions` thing already, I'm assuming that would be an object as a second argument. If so, I don't think that's currently part of the type signature of a test environment constructor, but I can definitely add it and it makes more sense to me. Just let me know what you'd prefer :)

Reply:
Here's the PR: #5003

Reply:
It's still on the config though, so I guess @cpojer's original objection still stands

Reply:
Ooooh, yeah, that's how my workaround works. I'm fine to do it either way. Though I think an extra argument might be better because using `testEnvironmentOptions` to add a `virtualConsole` would only make sense if the test environment is the jsdom one...

Reply:
We cannot add this as a config option.